tools/telemetry-generator/src/handlers/appInsightsExecutionTimeTestHandler.ts

```diff
@@ -33,30 +38,43 @@ module.exports = function handler(fileData, telemetryClient: TelemetryClient): v
 				},
 			});
 		} catch (error) {
-			console.error(`failed to emit metric ${arithmeticMeanMetricName}`, error);
+			console.error(`failed to emit metric '${arithmeticMeanMetricName}'`, error);
 		}
 
 		const marginOfErrorMetricName = `${fileData.suiteName}_${testData.benchmarkName}_marginOfError`;
 		try {
-			console.log(
-				`emitting metric ${arithmeticMeanMetricName} with value ${testData.customData["Margin of Error"]}`,
-			);
-			telemetryClient.trackMetric({
-				name: marginOfErrorMetricName,
-				value: testData.customData["Margin of Error"],
-				namespace: "performance_benchmark_executionTime",
-				properties: {
-					buildId: process.env.BUILD_ID,
-					branchName: process.env.BRANCH_NAME,
-					category: "performance",
-					eventName: "Benchmark",
-					benchmarkType: "ExecutionTime",
-					suiteName: fileData.suiteName,
-					benchmarkName: testData.benchmarkName,
-				},
-			});
+			const value = testData.customData["Margin of Error"];
+			if (value === undefined) {
+				console.error(`skipping metric '${marginOfErrorMetricName}' as it is undefined`);
+			} else {
+				// Remove ± and % characters
+				const parsedValue = Number.parseFloat(value.replace(/[%±]/g, ""));
```

Comment:
We shouldn't be editing the value in the handler, it should expect raw data. I'm also a bit confused because we didn't add this logic in the non-app-insights handler... would that one fail right now?

Reply:
We didn't. parsedValue used on validation only